### PR TITLE
automate compose-spec schema updates

### PR DESCRIPTION
**Pull request by ndeloof**
As suggested by https://github.com/compose-spec/compose-go/pull/339#issuecomment-1378039617
setup a cron job to sync compose-spec schema

### CODE REVIEW

File: .github/workflows/sync.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.3.0
+
+      - name: Download schema
+        run: wget --output-document=schema/compose-spec.json https://raw.githubusercontent.com/compose-spec/compose-spec/master/schema/compose-spec.json
+
+      - name: Show diff
+        run: git diff schema/compose-spec.json
+
```
Comment (glours) [inline .github/workflows/sync.yml:22]: Why not add those steps in the `validate` job of the CI workflow, this way having the spec in sync will be a prerequisite to merge a new PR, wdyt?

Comment (ndeloof) [reply .github/workflows/sync.yml:22]: would work as well. No preference

Comment (glours) [reply .github/workflows/sync.yml:22]: For me the main difference is to be sure compose-go spec version is always align with the spec everytime we merge a PR

**Review by glours — APPROVED** [review]
🥳 Thanks @ndeloof

**Pull request merged**